pkg/apis/camel/v1/trait/dependencies.go

```diff
@@ -20,7 +20,8 @@ package trait
 // The Dependencies trait is internally used to automatically add runtime dependencies based on the
 // integration that the user wants to run.
 //
-// +camel-k:trait=dependencies.
+// +camel-k:trait=dependencies
+// +camel-k:deprecated=2.4.0
```

Comment:
Why is the dependencies deprecated!?

Reply:
Maybe this should just get removed from the public docs i.e. it being internal

We could have ...

```
+camel-k:internal
```